projects/kit/components/input-date-range/input-date-range.template.html

```diff
@@ -11,7 +11,7 @@
         automation-id="tui-input-date-range__textfield"
         tuiValueAccessor
         class="t-textfield"
-        [tuiTextfieldIcon]="computedMobile ? mobileIconContent : iconContent"
+        [tuiTextfieldIcon]="hasCalendarIcon ? (computedMobile ? mobileIconContent : iconContent) : null"
```

Comment:
what do you think about implementing the pipe?

```html
[tuiTextfieldIcon]="calendarIcon | tuiPresent ? (computedMobile ? mobileIconContent : iconContent) : null"
```

Reply:
looks like a good idea

Reply:
There no need for a pipe like that, it only complicates things, besides empty `string` is also lack of icon, not just `null`. There's also no need for nesting a ternary operator, you can write it like that:
```html
[tuiTextfieldIcon]="computedMobile ? mobileIconContent : (calendarIcon && iconContent)"
```
If `calendarIcon` is falsy (empty `string` or `null`) it is passed to `tuiTextfieldIcon`, disabling the icon, otherwise `iconContent` is used.

Reply:
yes, but what about `mobileIconContent`? it should not be present as well

Reply:
besides, an empty string still renders `tui-svg`, it just has nothing to show. while `null` doesn't render anything

Reply:
while I agree that it looks more complicated, but using `null` instead of an empty string is a right decision imho. and then it comes down to this simple choice: method or pipe?

`hasCalendarIcon ? (computedMobile ? mobileIconContent : iconContent) : null`
vs
`(calendarIcon | tuiIsPresent) ? (computedMobile ? mobileIconContent : iconContent) : null`

Reply:
Ok. 
I don't see any difference between mobile and desktop icon content except for the click handler, but it has an early return anyway. How about we get rid of it and write:
```
[tuiTextfieldIcon]="calendarIcon && iconContent"
```
and change `iconContent` to:
```html
<ng-template #iconContent>
        <tui-svg
            *polymorpheusOutlet="calendarIcon as src; context: {$implicit: size}"
            automation-id="tui-input-date-range__icon"
            [class.t-icon]="!computedDisabled"
            [src]="src"
            (click)="onIconClick()"
        ></tui-svg>
</ng-template>
```
and just rename `onMobileClick` to `onIconClick` + don't forget to remove `open` toggle from the early return.

Reply:
As for the pipe, it would be a useful addition to the `cdk` but I don't think we really need it here.

Reply:
@waterplea do you want me to move the pipe to a separate PR?

Reply:
as for this one
```
[tuiTextfieldIcon]="calendarIcon && iconContent"
```
I still don't think that this is a right solution. if `calendarIcon` is an empty string, then the expression `'' && iconContent` becomes just `''` and you'll get an empty `tui-svg` rendered, as shown on the screenshot above. so the bug with an empty space next to a cleaner won't be fixed (which is a purpose of this PR)

Reply:
and I've just figured out that because of this very reason we don't need the pipe here 🙂
because it filters out only `nil` values, but we need an empty string to be removed as well

Reply:
how do you handle such cases in Taiga?

```ts
onIconClick(): void {
    if (!this.computedMobile) { // <-- this.isMobile && !!this.mobileCalendar
        return;
    }

    this.dialogService
        .open<TuiDay>(new PolymorpheusComponent(this.mobileCalendar, this.injector), {
        // 'Type<object> | null' is not assignable... ^^^^^^^^^^^^
}
```

I could add `!` to `mobileCalendar`, but it's not type safe, and the linter is also not happy with this solution

just check it again?

```ts
if (!this.computedMobile || !this.mobileCalendar) {
    return;
}
```